common/gamelet_component.cc

```diff
@@ -18,20 +18,28 @@
 
 #include "absl/strings/str_format.h"
 #include "absl/strings/str_split.h"
+#include "common/build_version.h"
 #include "common/path.h"
 #include "common/status.h"
 
 namespace cdc_ft {
 
-GameletComponent::GameletComponent(std::string filename, uint64_t size,
+GameletComponent::GameletComponent(std::string build_version,
+                                   std::string filename, uint64_t size,
                                    time_t modified_time)
-    : filename(filename), size(size), modified_time(modified_time) {}
+    : build_version(build_version),
+      filename(filename),
+      size(size),
+      modified_time(modified_time) {}
 
 GameletComponent::~GameletComponent() = default;
 
 bool GameletComponent::operator==(const GameletComponent& other) const {
-  return filename == other.filename && size == other.size &&
-         modified_time == other.modified_time;
+  return (build_version != DEV_BUILD_VERSION &&
```

Comment:
Nit: Maybe put the removed comment here, e.g.

// If either build version is the dev version, it means that the component was built locally, so that
// we can't compare build versions. Fall back to comparing file_size and modified_time.